lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -130,6 +134,22 @@ def add_debit(xml, money, options)
         end
       end
 
+      def add_preauth(xml, money, options)
+        currency    = options[:currency] || currency(money)
+        description = options[:description].blank? ? 'Preauthorize' : options[:description]
+
+        xml.preauthorize do
+          xml.transactionId new_transaction_id
+
+          add_customer_data(xml, options)
+
+          xml.amount      money
+          xml.currency    currency
+          xml.description description
+          xml.callbackUrl(options[:callback_url] || 'http://example.com')
```

Comment:
Are the parentheses here needed? (I have no idea.)

Reply:
I think Ruthan followed my example here. I used parentheses for the method call for clarity. Without them, I think the reader needs more knowledge of operator precedence to know whether the statement gets evaluated as

```xml.callbackUrl(options[:callback_url] || 'http://example.com')```
or 
```xml.callbackUrl(options[:callback_url]) || http://example.com```

Reply:
My thought was to write this line more like this:
`xml.callbackUrl   options[:callback_url] || 'http://example.com'`

Reply:
Yep. I mean that I find it ambiguous as to whether ```xml.callbackUrl options[:callback_url] || 'http://example.com'``` is equivalent to ```xml.callbackUrl(options[:callback_url] || 'http://example.com')``` or ```xml.callbackUrl(options[:callback_url]) || 'http://example.com'``` without the reader having good knowledge of Ruby's precedence evaluation.

Reply:
Hmm, okay. This seems quite straightforward to me, so I'm not seeing the confusion here. It's a pretty minor point, though, so either way is fine; I was mostly curious.

Reply:
After much contemplation, I initially opting in favor of the parens because I feel like they make the delineation between the method and the argument a little more legible. 

However, this made my attempt at combining the line formerly known as 138 with the line currently known as 148 look funny, so I've now tried something possibly worse. :D I'm not super attached to it, though, so whatever we decide we like best is good.

Reply:
Not to make your call on this harder, but I took the approach of omitting the parens on my `refund` PR that was approved by @jasonxp and has been merged: 
https://github.com/activemerchant/active_merchant/pull/3413/files#diff-51a89ecca18e8620ad6cb8e059787833R141

Barring that, though, I think that the approach you took is a good one @therufs.